invenio/legacy/search_engine/__init__.py

```diff
@@ -394,7 +433,7 @@ def check_user_can_view_record(user_info, recid):
         ## Perfect! It's authorized then!
         return (0, '')
 
-    restricted_collections = get_restricted_collections_for_recid(recid, recreate_cache_if_needed=False)
+    restricted_collections = get_restricted_collections_for_recid(recid, recreate_cache_if_needed=True)
```

Comment:
This is a module level cache, which means that the cache is not cleaned until the process is cleaned, hence you not only had to wait for webcoll/bibindex to run, you also had to wait for the process to be restarted.